src/gt4py/backend/module_generator.py

```diff
@@ -166,15 +172,23 @@ def make_args_data_from_gtir(pipeline: GtirPipeline, legacy=False) -> ModuleData
     ]
     for name in sorted(referenced_scalar_params):
         data.parameter_info[name] = ParameterInfo(
-            dtype=numpy.dtype(node.symtable_[name].dtype.name.lower())
+            access=AccessKind.READ, dtype=numpy.dtype(node.symtable_[name].dtype.name.lower())
         )
 
     unref_params = get_unused_params_from_gtir(pipeline)
     for param in sorted(unref_params, key=lambda decl: decl.name):
```

Comment:
It's somewhat strange that there are two seperate loops for the referenced and unreferenced params. If you are touching this part anyway you could combine the two and only use `get_unused_params_from_gtir`, but it's optional (as unrelated to the actual PR).